### PR TITLE
stft code and tests

### LINKED ISSUE #8827: stft
Adding stft signal function to tensorflow backend and frontend, and ivy functional.

### PR DESCRIPTION
**Pull request by abhayKashyap03**
Close #8827

### CODE REVIEW

**Comment by abhayKashyap03** [comment]
@YushaArif99 This is the new PR, please take a look at it and let me know if any other changes are needed.

**Comment by YushaArif99** [comment]
Hi @abhayKashyap03.  

You still seem to be modifying the `.idea/` files 😅.

**Comment by abhayKashyap03** [comment]
It looks like pycharm is making modifications to .idea folder everytime I use the application to make changes to the files. Just to make sure, so that I don't have to do this again and again, I just need to remove line number 9 from the file and add line number 12 back to the file right?

line 9 : ` <option name="format" value="NUMPY" />`
line 12 : ` <option name="projectConfiguration" value="pytest" />`

**Comment by abhayKashyap03** [comment]
@YushaArif99 @karalleyna Following up on the .idea changes, I've made the modifications I mentioned in the earlier comment and pushed the code, please check if everything's the way it's supposed to be and let me know if any other changes need to be made.

File: ivy/functional/ivy/experimental/layers.py [hunk 1]
```diff
@@ -770,3 +770,65 @@ def embedding(
         else:
             ret[i] = weights[x, :]
     return ret
+
+
+@handle_exceptions
+@to_native_arrays_and_back
+@handle_array_like
```
Comment (YushaArif99) [inline ivy/functional/ivy/experimental/layers.py:777]: This decorator has been renamed to `@handle_array_like_without_promotion`. Could you kindly make this change as currently all the CI tests are failing due to this error

**Review by YushaArif99 — CHANGES_REQUESTED** [review]
Hi @abhayKashyap03 

Thanks a lot for reverting the changes to `.idea/` files. A couple of points on the PR:
- Since you've also implemented the `stft` function for the TF frontend, could you also kindly add a test for it in the relevant testing file.
- Could you also make a small change I have requested above regarding the decorator so that the CI tests can run?
- There are also a few merge conflicts on your PR. Could you kindly fix them? You can basically edit them in the browser itself.

Overall, the changes do look good to me. However, could you first make these changes so I can verify that all relevant CI tests are passing.  
As always, let me know if you have any quesitons! 😃

File: ivy/functional/frontends/tensorflow/signal.py [hunk 2]
```diff
@@ -12,3 +12,15 @@ def kaiser_window(window_length, beta=12.0, dtype=ivy.float32, name=None):
 
 
 kaiser_window.supported_dtypes = ("float32", "float64", "float16", "bfloat16")
+
+
+@handle_tf_dtype
+@to_ivy_arrays_and_back
+def stft(signal, *, frame_length: Union[int, ivy.Array, ivy.NativeArray],
```
Comment (YushaArif99) [inline ivy/functional/frontends/tensorflow/signal.py:19]: You shouldn't use the same function formatting and type hints here that ivy functions use. The frontends' type hints should exactly match the type hints of the native function itself.
Therefore, in this case, the `stft's`  signature should be the same as `tf.signal.stft`

**Review by YushaArif99 — CHANGES_REQUESTED** [review]
Hi @abhayKashyap03 

Apologies for the late reply. Was swamped with work last week and so reviewing this PR totally slipped under my radar 😅 

Thanks for adding the frontend tests as well. There is a small error though in the frontend files that I have mentioned above. It is causing all CI tests to fail. Could you kindly look into it?

Also, for the sake of efficiency, I think its better to ensure that the relevant CI tests are all passing 
before requesting a review. The important tests are the `intelligent-tests-pr` and the `lint / check-formatting` tests. You can view the CI tests here (https://github.com/unifyai/ivy/actions/runs/4046653180/jobs/6959737483)

As always, let me know if you have any questions 😃

**Comment by abhayKashyap03** [comment]
@YushaArif99 I made the other changes you requested, but I don't know what I need to look for in the intelligent tests. It only says `exit 1` but doesn't give any other info on why it's failing. I've never had to fix this type of issue, so I don't know how to proceed with this. I googled the issue and what I can do to fix this but had no luck. Please help with what I can do to fix this.

**Comment by WilliamHirst** [comment]
Hi,
I have taken over this PR from Yasser.

It doesn’t seem like the intelligent-test are failing due to any changes made by you.

The first things to fix are the lint errors. These are simple formatting errors and should be relatively easy to fix.

The second thing to fix is the merge conflict. 

Make sure to fix these errors, then let me know! :)

**Comment by WilliamHirst** [comment]
There are still merge conflicts related to this PR. Please sort these out 😊

**Comment by abhayKashyap03** [comment]
@WilliamHirst I was just finishing up on the lint formatting test errors first, I will mention you and let you know once I'm done with all of the error/conflict resolutions. I've not been able to do much lately since I've had midterms, projects, and a lot of college work, but I'm going to get them all done as soon as I can and let you know.

**Pull request closed without merging**